### PR TITLE
added dynamic docker swarm + gitlab runner inventory

### PR DESCRIPTION
**Pull request by morph027**
##### SUMMARY

This adds a dynamic inventory for Docker swarm cluster nodes and Gitlab CI runners.

##### ISSUE TYPE

 - Feature Pull Request

##### ANSIBLE VERSION

```
2.3.0.0
```

### CODE REVIEW

**Comment by sivel** [comment]
At minimum, this script is missing the GPL3 license header

File: contrib/inventory/docker_swarm_nodes.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+class SwarmNodesInventory(object):
+
+    def __init__(self):
+        """ Main execution path """
+
+        # read settings
+        self.read_settings()
+
+        # initiate swarm api connection
+        import docker
```
Comment (Akasurde) [inline contrib/inventory/docker_swarm_nodes.py]: Might want to import at top level

File: contrib/inventory/docker_swarm_nodes.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    def read_settings(self):
+        """ Reads the settings from the docker_swarm_nodes.ini file """
+
+        config = ConfigParser.SafeConfigParser()
+        config.read(os.path.dirname(os.path.realpath(__file__)) + '/docker_swarm_nodes.ini')
+
+        self.docker_host = config.get('swarm', 'docker_host')
+
+SwarmNodesInventory()
```
Comment (Akasurde) [inline contrib/inventory/docker_swarm_nodes.py]: Nit pick - `if __name__ == "__main__"`

File: contrib/inventory/docker_swarm_nodes.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        # get worker nodes
+        for self.node in self.client.nodes.list(filters={'role': 'worker'}):
+            self.node_hostname = self.client.nodes.get(self.node.id).attrs['Description']['Hostname']
+            # append to hostgroup "swarm"
+            self.data["swarm"].append(self.node_hostname)
+            # append to hostgroup "worker"
+            self.data["worker"].append(self.node_hostname)
+
+        # finally print json
+        print json.dumps(self.data)
```
Comment (Akasurde) [inline contrib/inventory/docker_swarm_nodes.py]: print missing parentheses

**Review by Akasurde — CHANGES_REQUESTED** [review]

**Comment by morph027** [comment]
Thanks for the feedback. Being "just" a lazy sysadmin ansible user, my python is not the best ;)

**Pull request closed without merging**